queries.sql

```diff
@@ -7,4 +7,50 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
 SELECT name, date_of_birth FROM animals WHERE weight_kg >10.5;
 SELECT name FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT name FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT name FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+--  start transaction
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+SELECT species FROM animals;
+ROLLBACK;
+SELECT species FROM animals;
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name Like '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT sp1;
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO sp1;
+UPDATE animals
+SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+
+-- answer queries
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT name FROM animals 
```

Comment:
- [ ] Good job so far! Our query here does not however fully satisfy the question: _Who escapes the most, neutered or not neutered animals?_ We are unable know which animals are neutered or not. Let's therefore improve upon our query, which should be similar to the example below:
```sql
SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```